File: LeetCode/231.PowerOf2/Daniel.java

// Name: Daniel Eui Sung Kim
// Date: October 8th, 2017
// From: LeetCode
// Problem: #231 Power of Two

class Solution
{
    public boolean isPowerOfTwo(int n)
    {
        if ( n <= 0 ) return false;
        while ( n > 1 )
        {
            // power of 2 must return 0 at each division
            if ( n % 2 == 1 ) return false;
            n /= 2;
        }
        return true;
    }
}

File: LeetCode/001.TwoSum/Daniel.cpp

// Name: Daniel Eui Sung Kim
// Date: October 12th, 2017
// From: LeetCode
// Problem: #1 Two Sum

class Solution
{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        vector<int> results;
        unordered_map<int, int> map;
        for ( auto i = 0; i < nums.size(); i++ )
        {
            // if "two sum" found
            if ( map.count(target - nums[i]) )
            {
                // add index
                results.push_back(map[target - nums[i]]);
                results.push_back(i);
            }
            // else, nums[i] = key, index = value.
            map[nums[i]] = i;
        }
        
        return results;
    }
    
};